src/CommunityToolkit.Aspire.Hosting.Python.Extensions/UvAppHostingExtension.cs

```diff
@@ -0,0 +1,99 @@
+    /// </summary>
+    /// <param name="builder">The distributed application builder.</param>
+    /// <param name="name">The name of the Uv application.</param>
+    /// <param name="projectDirectory">The directory of the project containing the Uv application.</param>
+    /// <param name="scriptPath">The name of the Uv app.</param>
+    /// <param name="scriptArgs">Optional arguments to pass to the script.</param>
+    /// <returns>An <see cref="IResourceBuilder{UvAppResource}"/> for the Uv application resource.</returns>
+    /// <exception cref="ArgumentNullException">Thrown if <paramref name="builder"/> is null.</exception>
+    public static IResourceBuilder<UvAppResource> AddUvApp(
+        this IDistributedApplicationBuilder builder,
+        string name,
+        string projectDirectory,
+        string scriptPath,
+        params string[] scriptArgs)
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+
+        return builder.AddUvApp(name, scriptPath, projectDirectory, ".venv", scriptArgs);
+    }
+
+    private static IResourceBuilder<UvAppResource> AddUvApp(this IDistributedApplicationBuilder builder,
+        string name,
+        string scriptPath,
+        string projectDirectory,
+        string virtualEnvironmentPath,
+        params string[] args)
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+        ArgumentNullException.ThrowIfNull(name);
+        ArgumentNullException.ThrowIfNull(scriptPath);
+
+        string wd = projectDirectory ?? Path.Combine("..", name);
+
+        projectDirectory = PathNormalizer.NormalizePathForCurrentPlatform(Path.Combine(builder.AppHostDirectory, wd));
+
+        var virtualEnvironment = new VirtualEnvironment(Path.IsPathRooted(virtualEnvironmentPath)
+            ? virtualEnvironmentPath
+            : Path.Join(projectDirectory, virtualEnvironmentPath));
+
+        var instrumentationExecutable = virtualEnvironment.GetExecutable("opentelemetry-instrument");
+        // var pythonExecutable = virtualEnvironment.GetRequiredExecutable("python");
+        // var projectExecutable = instrumentationExecutable ?? pythonExecutable;
+
+        string[] allArgs = args is { Length: > 0 }
+            ? ["run", scriptPath, .. args]
+            : ["run", scriptPath];
+
+        var projectResource = new UvAppResource(name, projectDirectory);
+
+        var resourceBuilder = builder.AddResource(projectResource)
+            .WithArgs(allArgs)
+            .WithArgs(context =>
+            {
+                // If the project is to be automatically instrumented, add the instrumentation executable arguments first.
+                if (!string.IsNullOrEmpty(instrumentationExecutable))
+                {
+                    AddOpenTelemetryArguments(context);
+
+                    // // Add the python executable as the next argument so we can run the project.
+                    // context.Args.Add(pythonExecutable!);
+                }
+            });
```

Comment:
may as well add this to the below `if` block and just have it `resourceBuilder.WithArgs(AddOpenTelemetryArguments);` since the `if` block is duplicated.

Reply:
I was following the same logic found in https://github.com/dotnet/aspire/blob/d7e27b157b0266fc17a116115ba55d773684eba6/src/Aspire.Hosting.Python/PythonAppResourceBuilderExtensions.cs#L131

Reply:
ah right, let's leave it